tests/unit/s2n_ecc_evp_test.c

```diff
@@ -345,5 +361,52 @@ int main(int argc, char **argv) {
         }
     }
 
+    /* Test that the client does not negotiate a group that was not
+     * offered in EC preferences */
+    {
+        const struct s2n_security_policy* security_policy = NULL;
+        DEFER_CLEANUP(struct s2n_connection* conn = s2n_connection_new(S2N_CLIENT),
+        s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(conn);
+        EXPECT_SUCCESS(s2n_connection_set_cipher_preferences(conn, "20190802"));
+        EXPECT_SUCCESS(s2n_connection_get_security_policy(conn, &security_policy));
+
+        /* Setup & verify invalid curves, which will be selected by a malicious server */
+        const struct s2n_ecc_named_curve* const unrequested_curves[] = {
+                &s2n_unsupported_curve,
+                &s2n_ecc_curve_secp521r1,
+        };
+
+        /* Verify that the client broadcasts an error code when the server attempts to
+         * negotiate a curve that was never offered */
```

Comment:
```suggestion
        /* Verify that the client errors when the server attempts to negotiate a curve that was never offered */
```
"broadcasts an error code" sounds a little off, since we're just setting a thread local variable :)